File: ReactNative-version/frontend/src/domain/life-sphere.ts
/** IDs from MASTER_PLAN §2. Add a sphere here; UI maps labels separately. */
export const LIFE_SPHERES = [
  'work',
  'body',
  'social',
  'rest',
  'home',
  'growth',
] as const;

export type LifeSphereId = (typeof LIFE_SPHERES)[number];

export function isLifeSphereId(value: string): value is LifeSphereId {
  return (LIFE_SPHERES as readonly string[]).includes(value);
}

File: flutter-version/lib/domain/life_sphere.dart
/// IDs match MASTER_PLAN §2 (`work`, `body`, …).
enum LifeSphere {
  work,
  body,
  social,
  rest,
  home,
  growth,
}

extension LifeSphereStorage on LifeSphere {
  String get id => name;

  static LifeSphere? tryParse(String? raw) {
    if (raw == null || raw.isEmpty) return null;
    for (final s in LifeSphere.values) {
      if (s.id == raw) return s;
    }
    return null;
  }

  static LifeSphere parseOrWork(String raw) {
    return tryParse(raw) ?? LifeSphere.work;
  }
}
